ci/build.ps1

```diff
@@ -38,15 +41,18 @@ if ($compiler -eq 'MINGW') {
   # These are native MinGW builds, but they use the toolchain inside
   # MSYS2, this allows using all the dependencies and tools available
   # in MSYS2, but we cannot build inside the MSYS2 shell.
-  $cmakeGenerator = 'MinGW Makefiles'
-  $cmakeGeneratorArgs = 'VERBOSE=1'
+  $cmakeGenerator = 'Ninja'
+  $cmakeGeneratorArgs = '-v'
+  $mingwPackages = @('ninja', 'gcc', 'make', 'cmake', 'perl', 'diffutils', 'unibilium').ForEach({
+    "mingw-w64-$arch-$_"
+  })
```

Comment:
Did you add `gcc` to upgrade gcc? If possible we should use whatever appveyor already has, otherwise it adds variability to the builds.

Reply:
I added it in the event I run this script on Windows like #8405 but it's better to update Makefile for Windows.